File: idakonyvtar/src/main/java/eu/eyan/idakonyvtar/view/LibraryView.java
package eu.eyan.idakonyvtar.view;

import java.awt.Component;

import javax.swing.JPanel;
import javax.swing.JScrollPane;

import com.jgoodies.forms.factories.CC;
import com.jgoodies.forms.layout.FormLayout;

public class LibraryView extends AbstractView {
	private final BookTable bookTable = new BookTable();

	@Override
	public Component createViewComponent() {
		JPanel panel = new JPanel(
				new FormLayout("pref:grow", "pref, pref:grow"));
		JScrollPane scrollPane = new JScrollPane(getBookTable());
		panel.add(scrollPane, CC.xy(1, 2));
		return scrollPane;
	}

	public BookTable getBookTable() {
		return bookTable;
	}
}

File: idakonyvtar/src/main/scala/eu/eyan/idakonyvtar/view/LibraryView.scala
package eu.eyan.idakonyvtar.view;

import java.awt.Component
import javax.swing.JPanel
import javax.swing.JScrollPane
import com.jgoodies.forms.factories.CC
import com.jgoodies.forms.layout.FormLayout;
import eu.eyan.util.swing.JPanelWithFrameLayout

class LibraryView extends AbstractView {
  private val bookTable = new BookTable
  def getBookTable() = bookTable

  override def createViewComponent(): Component = {
    val panel = new JPanelWithFrameLayout()
      .withBorders
      .withSeparators
      .newColumnFPG
      .newRowFPG
      
    val scrollPane = new JScrollPane(getBookTable())
    panel.add(scrollPane)
    panel
  }
}